governor/tests/keyed_dashmap.rs

```diff
@@ -46,6 +46,215 @@ fn rejects_too_many() {
     }
 }
 
+#[test]
+fn rejects_too_many_n() {
+    let clock = FakeRelativeClock::default();
+    let lb = RateLimiter::dashmap_with_clock(Quota::per_second(nonzero!(2u32)), &clock);
+    let ms = Duration::from_millis(1);
+
+    for key in KEYS {
+        // use up our burst capacity (2 in the first second):
+        assert!(
+            lb.check_key_n(key, nonzero!(2u32)).is_ok(),
+            "Now: {:?}",
+            clock.now()
+        );
+        clock.advance(ms);
+
+        assert_ne!(Ok(()), lb.check_key(key), "Now: {:?}", clock.now());
+
+        // should be ok again in 1s:
+        clock.advance(ms * 1000);
+        assert!(
+            lb.check_key_n(key, nonzero!(2u32)).is_ok(),
+            "Now: {:?}",
+            clock.now()
+        );
```

Comment:
ditto on coverage of that multiline-assert.

Reply:
Yeah, took a while, but I think I got it now. If I missed anything, please let me know.